### PR TITLE
[sparse] sparse.nn.functional.dropout()

### PR DESCRIPTION
**Pull request by weiyangfb**
- add `sparse.nn.functional.dropout` with autograd support

### CODE REVIEW

**Comment by vadimkantorov** [comment]
@soumith another mini-clash with namespaces. `torch.nn.functional.dropout` and `torch.sparse.dropout`

**Comment by soumith** [comment]
@vadimkantorov it's intentional. `sparse.xxx` have a different meaning for what gradients they compute.

**Comment by vadimkantorov** [comment]
@soumith I understand the semantic difference, I meant just the naming "nn.functional.xxx" vs "sparse.xxx" (and not "sparse.nn.functional.xx")

**Comment by soumith** [comment]
ah yes, sorry I misunderstood.

**Comment by weiyangfb** [comment]
sure, just moved it to `torch.sparse.nn.functional.dropout`

**Comment by vadimkantorov** [comment]
( @weiyangfb `torch.sparse.dropout` unfixed in docstring )
